File: ui_v2/backend/models/bot.py
"""Pydantic models for bot metadata, status, and config."""

from typing import Any, Dict, Optional
from pydantic import BaseModel


class BotInfo(BaseModel):
    id: str
    name: str
    description: str
    instructions: str = ""
    tier: str = "?"
    file_path: str
    module_name: str
    config_params: Dict[str, Any] = {}
    default_config: Dict[str, Any] = {}


class BotStatusResponse(BaseModel):
    status: str  # running | paused | terminated | not_running
    paused: Optional[bool] = None
    runtime: Optional[float] = None
    runtime_formatted: Optional[str] = None
    start_time: Optional[float] = None


class BotStartRequest(BaseModel):
    config: Dict[str, Any] = {}
    username: str = ""


class BotControlRequest(BaseModel):
    action: str  # pause | resume | terminate
    value: Optional[Any] = None


class SuccessResponse(BaseModel):
    success: bool
    error: Optional[str] = None

File: ui_v2/frontend/src/types/bot.ts
import type { BotConfig, ConfigParam } from './config';

/** Bot info as returned by GET /api/bots */
export interface BotInfo {
  id: string;
  name: string;
  description: string;
  instructions: string;
  tier: string;
  file_path: string;
  module_name: string;
  config_params: Record<string, ConfigParam>;
  default_config: Record<string, unknown>;
}

/** Maps bot_id → BotInfo */
export type BotRegistry = Record<string, BotInfo>;

/** GET /api/bot/{id}/status */
export interface BotStatus {
  status: 'running' | 'paused' | 'terminated' | 'not_running';
  paused?: boolean;
  runtime?: number;
  runtime_formatted?: string;
  start_time?: number;
}

/** POST /api/bot/{id}/start body */
export interface BotStartRequest {
  config: Record<string, unknown>;
  username: string;
}

/** POST /api/bot/{id}/control body */
export interface BotControlRequest {
  action: 'pause' | 'resume' | 'terminate';
  value?: unknown;
}

/** Generic success response */
export interface SuccessResponse {
  success: boolean;
  error?: string;
}
